unittest2pytest/fixes/fix_remove_class.py

```diff
@@ -57,17 +57,28 @@ class FixRemoveClass(BaseFix):
     """
 
     def dedent(self, suite, dedent):
-        self.line_num = suite.get_lineno()
-        for kid in suite.leaves():
-            if kid.type in (token.INDENT, token.DEDENT):
-                self.line_num = kid.get_lineno()
-                # todo: handle tabs
-                kid.value = kid.value[dedent:]
-                self.current_indent = kid.value
-            elif kid.get_lineno() != self.line_num:
-                # todo: handle tabs
-                if len(kid.prefix) > len(self.current_indent):
-                    kid.prefix = self.current_indent
+        def safe_dedent(s):
+            for i, c in enumerate(s):
+                if c not in "\r\n":
+                    break
+            else:
+                i = len(s)
+            return s[:i] + s[i:-dedent]
+
+        leaves = suite.leaves()
+        for leaf in leaves:
+            if leaf.type == token.NEWLINE:
+                leaf = next(leaves, None)
+                if leaf is None:
+                    return
+                if leaf.type == token.INDENT:
+                    leaf.value = safe_dedent(leaf.value)
+                else:
+                    leaf.prefix = safe_dedent(leaf.prefix)
+            elif leaf.type == token.INDENT:
+                leaf.value = leaf.value[:-dedent]
+            elif leaf.prefix[:1] in "\r\n":
```

Comment:
```suggestion
            elif leaf.startswith(("\r", "\n")):
```

Reply:
both are safe, and it turns out the second form is ~50% slower:
```
$ python -m timeit '""[:1] in "ab"'
5000000 loops, best of 5: 48.6 nsec per loop
$ python -m timeit '"".startswith(("a", "b"))'
5000000 loops, best of 5: 70.5 nsec per loop
$ python -m timeit '"b"[:1] in "ab"'
5000000 loops, best of 5: 49.8 nsec per loop
$ python -m timeit '"b".startswith(("a", "b"))'
5000000 loops, best of 5: 77.8 nsec per loop
$ python -m timeit --setup="s = 'x'*1000" 's[:1] in "ab'
5000000 loops, best of 5: 48.4 nsec per loop
$ python -m timeit --setup="s = 'x'*1000" 's.startswith(("a", "b"))'
5000000 loops, best of 5: 75.7 nsec per loop
```

Reply:
Anyhow, the later form is much more easy to understand. And there is not need to optimize for speed in this project, isn't it?